config/config-api/src/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -414,8 +421,17 @@ private void verifyUniqueNameInParts(PipelineConfig pipelineConfig) {
 
         @Override
         public void update(String groupName, String pipelineName, PipelineConfig pipeline) {
-            // this was called only from tests
-            throw bomb("Cannot update pipeline group in merged configuration");
+            if (groups.isEmpty()) {
```

Comment:
Why do we need this to be implemented for `MergeStrategy`? As a part of https://github.com/gocd/gocd/pull/2256, I am getting rid of the update method on MergeStrategy. Am I missing any scenario?